notebook/static/notebook/js/keyboardmanager.js

```diff
@@ -42,6 +42,26 @@ define([
         this.edit_shortcuts = new keyboard.ShortcutManager(undefined, options.events, this.actions, this.env);
         this.edit_shortcuts.add_shortcuts(this.get_default_common_shortcuts());
         this.edit_shortcuts.add_shortcuts(this.get_default_edit_shortcuts());
+
+
+        this.config = options.config;
+        var that = this;
+
+        this.config.loaded.then(function(){ 
+            
+            (((that.config.data.keys||{}).edit||{})
```

Comment:
This is extremely dense, and I can't really follow where the different parentheses start and stop. Maybe expand it a little bit?